File: C++/OOP Lab4/exceptions.cpp
#include "exceptions.h"

const char* MyBaseException::what() const noexcept {
	return "MyBaseException occured";
}

const char* ChoiceError::what() const noexcept {
	return "Invalid choice";
}

const char* ListException::what() const noexcept {
	return "Mushroom list is empty";
}

File: Python/exceptions.py
class MyBaseException(Exception):
    def __init__(self, error_message=None):
        self.error_message = error_message

    def __str__(self):
        return f"MyBaseException\n{self.error_message}"


class ChoiceError(MyBaseException):
    def __str__(self):
        return "Invalid choice"


class ListException(MyBaseException):
    def __str__(self):
        return "Mushroom list is empty!"
